### DISCUSSION TITLE
VERCEL_URL environment variable doesn't work

### DISCUSSION THREAD

**Discussion by cgrgnc** [vercel/community#729] [Help] [answered]
Hi, I want to add a condition one of my URLs in vercel. I'm using Next.js and my code is below

index.tsx
```
const Home: NextPage = () => {
  const [textEnv, setTextEnv] = useState<string>()
  const [textUrl, setTextUrl] = useState<string>()

  useEffect(()=>{
    if(process.env.NEXT_PUBLIC_VERCEL_ENV == "development"){
      setTextEnv("this is from development")
    }else if(process.env.NEXT_PUBLIC_VERCEL_ENV == "production"){
      setTextEnv("this is from production")
    }else if(process.env.NEXT_PUBLIC_VERCEL_ENV == "preview"){
      setTextEnv("this is from preview")
    }else{
      setTextEnv("nothing else")
    }
    if(process.env.NEXT_PUBLIC_VERCEL_URL === "vercel-env-test-seven.vercel.app"){
      setTextUrl("this the url what I look for")
    }else{
      setTextUrl("other urls")
    }
  },[])
  return (
    <div>
      <Box display="flex" flexDirection="column">
        <Text>Type of Environment</Text>
        <Text>{textEnv}</Text>
        <Text>Name of the URL: </Text>
        <Text>{textUrl}</Text>
      </Box>
    </div>
  )
}

```

The result webpage: vercel-env-test-seven.vercel.app
 
 I checked my project settings. Automatically expose System Environment Variables is ticked.
 I tried also only VERCEL_URL but it doesn't work.
 NEXT_PUBLIC_VERCEL_ENV works, but VERCEL_URL doesn't work.
 Can you help me where the problem is?

**Comment by swarnava** [comment, marked as answer]
Can you create a [minimal step ](https://stackoverflow.com/help/minimal-reproducible-example)to reproduce this so that we can take a look?

**Reply by cgrgnc** [reply]
I just added.

**Reply by amyegan** [reply]
Thanks for updating with your code and reproduction steps! It looks like you're accessing the environment variable in client-side code, so it makes sense that you can access a `NEXT_PUBLIC_` prefixed variable and not other environment variables there. The framework prefix is what is making the variable available to the browser.

**Reply by uziel-bueno** [reply]
Hi @amyegan , I'm trying to reference the system environment `NEXT_PUBLIC_VERCEL_URL` from a custom env variable, but unfortunately it is not working, do you think I'm missing something?

<img width="983" alt="image" src="https://user-images.githubusercontent.com/77310379/197416797-bc540d1e-a2a8-4e0a-89d8-7d283b4e868b.png">

<img width="505" alt="image" src="https://user-images.githubusercontent.com/77310379/197416938-f820b481-dc61-4dca-a199-9e01d69e8b46.png">

**Reply by amyegan** [reply]
Environment variable expansion does not currently have native support from the Environment Variables UI, so you will not be able to access `NEXT_PUBLIC_VERCEL_URL` from a custom env var. Next.js will automatically expand variables inside `.env*` files, but I don't believe the feature works with UI managed variables at the moment.

**Reply by uziel-bueno** [reply]
@amyegan Thanks for replying! I see, then, do you suggest to add a `.env.development` file with the following content?

NEXT_PUBLIC_BASE_URL=${NEXT_PUBLIC_VERCEL_URL}
